libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -429,11 +429,13 @@ - (void)logoutUser:(SFUserAccount *)user
         return;
     }
 
-    if (user.isUserLoggingOut) {
-        [self log:SFLogLevelInfo msg:@"logoutUser: user is already in the process of logout."];
-        return;
+    @synchronized (user) {
```

Comment:
I am working off little sleep atm, but this might be something I need to get down on paper to see if any edge cases pop out. I will look again tomorrow.